synapse/handlers/federation.py

```diff
@@ -1452,6 +1457,20 @@ async def get_room_complexity(
         # well.
         return None
 
+    async def _resume_sync_partial_state_room(self) -> None:
+        """Resumes resyncing of all partial-state rooms after a restart."""
+        assert not self.config.worker.worker_app
+
+        partial_state_rooms = await self.store.get_partial_state_rooms_and_servers()
+        for room_id, servers_in_room in partial_state_rooms.items():
+            run_as_background_process(
+                desc="sync_partial_state_room",
+                func=self._sync_partial_state_room,
+                destination=None,
+                destinations=servers_in_room,
+                room_id=room_id,
+            )
```

Comment:
I could not see a simple way to determine the `destination` the original sync was using after a restart, so I took the entire list of servers in the room to try.

Reply:
yeah, I think this is ok for now, but it definitely leads to less than satisfactory results while it works its way through the entire server list of Matrix HQ trying to find a server which will respond.